Ollivanders/src/net/pottercraft/Ollivanders2/Book/Book.java

```diff
@@ -95,19 +96,17 @@ private void writeSpellBookMeta (ItemStack bookItem)
       {
          bookContents.add(spell.toString());
       }
-      for (String potion : potions)
-      {
-         bookContents.add(potion);
-      }
+
+      bookContents.addAll(potions);
 
       for (String content : bookContents)
       {
-         String name = BookTexts.getName(content);
+         String name = p.books.spellText.getName(content);
```

Comment:
Where is does p get set to something besides null?

Reply:
Oh good catch, it needs to be passed in to the constructor for Books

Reply:
Added it to the constructor for Book